scripts/src/javatests/org/oppia/android/scripts/maven/GenerateMavenDependenciesListTest.kt

```diff
@@ -0,0 +1,423 @@
+    tempFolder.newFile("scripts/assets/maven_dependencies.textproto")
+    val pbFile = tempFolder.newFile("scripts/assets/maven_dependencies.pb")
+    val license1 = License.newBuilder().apply {
+      this.licenseName = "The Apache License, Version 2.0"
+      this.originalLink = "https://www.apache.org/licenses/LICENSE-2.0.txt"
+      this.scrapableLink = ScrapableLink.newBuilder()
+        .setUrl("https://www.apache.org/licenses/LICENSE-2.0.txt").build()
+    }.build()
+    val license2 = License.newBuilder().apply {
+      this.licenseName = "Simplified BSD License"
+      this.originalLink = "https://www.opensource.org/licenses/bsd-license"
+      this.localCopyLink = LocalCopyLink.newBuilder()
+        .setUrl("https://local-copy/bsd-license").build()
+    }.build()
+    val mavenDependencyList = MavenDependencyList.newBuilder().apply {
+      this.addAllMavenDependency(
+        listOf(
+          MavenDependency.newBuilder().apply {
+            this.artifactName = DATA_BINDING_COORD
+            this.artifactVersion = DATA_BINDING_COORD_VERSION
+            this.addAllLicense(listOf(license1))
+          }.build(),
+          MavenDependency.newBuilder().apply {
+            this.artifactName = PROTO_LITE_COORD
+            this.artifactVersion = PROTO_LITE_COORD_VERSION
+            this.addAllLicense(listOf(license2))
+          }.build()
+        )
+      )
+    }.build()
+    mavenDependencyList.writeTo(pbFile.outputStream())
+
+    val coordsList = listOf(DATA_BINDING_COORD, PROTO_LITE_COORD)
+    setupBazelEnvironment(coordsList)
+
+    GenerateMavenDependenciesListRunner(
+      mockLicenseFetcher,
+      CommandExecutorImpl(600_000L)
+    ).main(
+      arrayOf(
+        "${tempFolder.root}",
+        "scripts/assets/maven_install.json",
+        "${tempFolder.root}/scripts/assets/maven_dependencies.pb"
+      )
+    )
+    assertThat(outContent.toString()).contains(SCRIPT_PASSED_MESSAGE)
+  }
+
+  private fun setupBazelEnvironment(coordsList: List<String>) {
+    val mavenInstallJson = tempFolder.newFile("scripts/assets/maven_install.json")
+    writeMavenInstallJson(mavenInstallJson)
+    testBazelWorkspace.ensureWorkspaceIsConfiguredForRulesJvmExternal(coordsList)
+    val thirdPartyPrefixCoordList = coordsList.map { coordinate ->
+      "//third_party:${omitVersionAndReplaceColonsHyphensPeriods(coordinate)}"
+    }
+    createAndroidBinary(thirdPartyPrefixCoordList)
+    writeThirdPartyBuildFile(coordsList)
+  }
+
+  private fun writeThirdPartyBuildFile(exportsList: List<String>) {
+    val thirdPartyBuild = tempFolder.newFile("third_party/BUILD.bazel")
+    thirdPartyBuild.appendText(
+      """
+      load("@rules_jvm_external//:defs.bzl", "artifact")
+      """.trimIndent() + "\n"
+    )
+    for (export in exportsList) {
+      createAndroidLibrary(thirdPartyBuild, export)
+    }
+  }
+
+  private fun createAndroidLibrary(thirdPartyBuild: File, artifactName: String) {
```

Comment:
Perhaps ``createThirdPartyAndroidLibrary`` would be clearer?

Reply:
Done.

Reply:
Doesn't appear done yet.

Reply:
Sorry, didn't push earlier. Now done.

Reply:
In the latest version I still see ``createAndroidLibrary``

Reply:
Apologies, it's done now.